src/v/rpc/transport.cc

```diff
@@ -52,15 +52,14 @@ struct client_context_impl final : streaming_context {
 };
 
 transport::transport(
-  transport_configuration c,
-  [[maybe_unused]] std::optional<ss::sstring> service_name)
+  transport_configuration c, std::optional<connection_cache_label> label)
```

Comment:
Indeed, the connection cache is a higher level abstraction. The idea behind the label is to identify to which connection_cache instance a connection belongs to. Previously, we created connection metrics with labels for the target host and port. That worked fine with one connection cache, but can lead to double registrations when there's multiple connection_cache instances.

I don't know if this means that we should lift the connection level metrics. I think they're in the right place tbh. Maybe the weirdness comes from the naming mismatch (i.e. if `transport` was called `connection` it would be more clear that the label indicates ownership).

Reply:
thanks for the explanation. i think it's probably ok for now